src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1228,13 +1229,15 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
     }
 
     FileOutErr outErr = action.getSpawnExecutionContext().getFileOutErr();
-
-    // Always download the action stdout/stderr.
     FileOutErr tmpOutErr = outErr.childOutErr();
-    List<ListenableFuture<Void>> outErrDownloads =
-        remoteCache.downloadOutErr(context, result.actionResult, tmpOutErr);
-    for (ListenableFuture<Void> future : outErrDownloads) {
-      downloadsBuilder.add(transform(future, (v) -> null, directExecutor()));
+
+    // Skip downloading the action stdout/stderr when remote_download_outputs=minimal and the action is successful.
```

Comment:
should this apply to `remote_download_outputs=toplevel` too?

Reply:
I think that would be a more complicated change that could be addressed in a separate PR. I want to keep this PR simple to discuss the right UX first.